lib/msf/core/payload/stager.rb

```diff
@@ -23,6 +26,17 @@ def initialize(info={})
   end
 
   #
+  # Perform attempt at detecting the appropriate transport config.
+  # Call the determined config with passed options.
+  # Override this in stages/stagers to use specific transports
+  #
+  def transport_config(opts={})
+    transport_name = "transport_config_#{self.refname =~ /reverse_/ ? 'reverse' : 'bind'}" +
+      "_#{self.refname =~ /_tcp/ ? 'tcp' : 'http'}"
+    send(transport_name.to_sym,opts)
```

Comment:
We need to cover the `https` case as well. Perhaps to make it a little more readable, we could go with the simpler:
``` ruby
if self.refname ~= /reverse_/
    direction = 'reverse'
else
    direction = 'bind'
end

if self.refname =~ /_tcp/
    proto = 'tcp'
elsif self.refname =~ /_https/
    proto = 'https'
else
    proto = 'http'
end
send("transport_config_#{direction}_#{proto}", opts)
```
I like the idea in the short term though, this will give us the coverage we need while we're porting the stagers to metasm.